### PR TITLE
Updated README.rst Example to be a Regressor

### PR DESCRIPTION
**Pull request by ZiadAmerr**
This PR is linked to [this trello card](https://trello.com/c/FwXlT3zJ/1506-improve-example-in-the-readmerst-file).

### CODE REVIEW

**Comment by kurshakuz** [comment]
@ZiadAmerr let me know once it is ready for review

**Comment by ZiadAmerr** [comment]
Hey @kurshakuz, it is ready now :)
I will just request a review from @guillesanbri before we continue with the merge

File: README.rst [hunk 1]
```diff
@@ -229,33 +231,52 @@ but this can easily be changed to your favorite framework, such as TensorFlow, o
...
+    # generate some random data
+    x = ivy.random.random_normal(shape=(100, 3))
+    y = ivy.random.random_normal(shape=(100, 1))
 
-    for step in range(100):
-        loss, grads = ivy.execute_with_gradients(loss_fn, model.v)
+    def loss_fn(pred, target):
+        return ivy.mean((pred - target)**2)
+
+    for epoch in range(51):
```
Comment (guillesanbri) [inline README.rst:258]: very minor thing but maybe it's better to leave this as `50` and instead add one in the print statement if it's due to formatting of the results?

**Review by guillesanbri — APPROVED** [review]
Very minor thing, otherwise lgtm, feel free to merge :)

**Pull request merged**